Bottom: 274113cca73ecedc8e41e9820d44edeaba0125b1
Top:    44bfde289ab26aeb4198759eb1b330a2100c5203
Author: Jason Wang <jasowang@redhat.com>
Date:   2016-05-12 09:17:47 +0800

Refresh of high-performance-mode

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index ffe0ff2..4eb608c 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -928,8 +928,6 @@ static netdev_tx_t tun_net_xmit(struct sk_buff *skb, struct net_device *dev)
 		spin_lock_irqsave(&tfile->wlock, flags);
 
 		head = tfile->head;
-		/* The spin_unlock() and next spin_lock() provide
-		 * needed ordering. */
 		tail = ACCESS_ONCE(tfile->tail);
 
 		if (CIRC_SPACE(head, tail, TUN_RING_SIZE) >= 1) {
